System/Console/ANSI/Types.hs

```diff
@@ -26,11 +26,29 @@ data Color = Black
            | White
            deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
--- | ANSI colors come in two intensities
+-- | ANSI's standard colors come in two intensities.
 data ColorIntensity = Dull
                     | Vivid
                     deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
+-- | Represents xterm's protocol for a 256-color palette.
+--
+-- The protocol has three parts:
+--
+-- (1) the eight standard, or 'system', colors in dull and vivid intensities;
+--
+-- (2) colors in RGB space, with the red, green and blue color components being
+-- specified by a integer from 0 to 5; and
+--
+-- (3) a spectrum of 24 grays, from dark gray to near white (black and white are
+-- themselves excluded).
+data Xterm256Color
+  = SystemColor !ColorIntensity !Color -- ^ The ANSI standard colors
+  | RGBColor !(Colour Float)           -- ^ Colors in RGB space
+  | GrayColor !Float                   -- ^ Grayscale (input is expected to be
+                                       -- in the range [0..1])
```

Comment:
we could add something like "and the function that converts it to its discrete xterm representation is ..." for the same reason as my comment above

Reply:
No longer applicable in my new revised approach, but I have added a lot more documentation, including cross-references and examples.